src/predict.jl

```diff
@@ -50,8 +50,18 @@ function predict(
 
     log_debug(verbosity, "Started predicting\n")
 
+    if num_threads != -1
```

Comment:
I moved the logic for `num_threads` here and pass it together with parameters so there is no need to add an extra parameter to the wrapper for `LGBM_BoosterPredictForMat`. The initial condition was `if num_threads > 0` and only in such case this parameter was passed to LGBM_BoosterPredictForMat, otherwise it wasn't specified. I now made it so that it's always passed in parameters and either is what user provided in `predict()`or from estimator.